anesthetic/plot.py

```diff
@@ -359,49 +358,111 @@ def contour_plot_2d(ax, data_x, data_y, *args, **kwargs):
     label = kwargs.pop('label', None)
     zorder = kwargs.pop('zorder', 1)
+    linewidths = kwargs.pop('linewidths', 0.5)
+    levels = kwargs.pop('levels', [0.68, 0.95])
     color = kwargs.pop('color', next(ax._get_lines.prop_cycler)['color'])
+
     if len(data_x) == 0 or len(data_y) == 0:
         return numpy.zeros(0), numpy.zeros(0), numpy.zeros((0, 0))
 
     x, y, pdf = kde_2d(data_x, data_y,
                        xmin=xmin, xmax=xmax, ymin=ymin, ymax=ymax)
-    pdf /= pdf.max()
-    p = sorted(pdf.flatten())
-    m = numpy.cumsum(p)
-    m /= m[-1]
-    interp = interp1d([0]+list(m)+[1], [0]+list(p)+[1])
-    contours = list(interp([0.05, 0.33]))+[1]
-
-    # Correct non-zero edges
-    if min(p) != 0:
-        contours = [min(p)] + contours
-
-    # Correct level sets
-    for i in range(1, len(contours)):
-        if contours[i-1] == contours[i]:
-            for j in range(i):
-                contours[j] = contours[j] - 1e-5
-
-    i = (pdf >= 1e-2).any(axis=0)
-    j = (pdf >= 1e-2).any(axis=1)
 
+    levels = iso_probability_contours(pdf, contours=levels)
     cmap = basic_cmap(color)
 
-    cbar = ax.contourf(x[i], y[j], pdf[numpy.ix_(j, i)], contours, cmap=cmap,
-                       zorder=zorder, vmin=0, vmax=1.0, *args, **kwargs)
+    i = (pdf >= levels[0]*0.5).any(axis=0)
+    j = (pdf >= levels[0]*0.5).any(axis=1)
+
+    cbar = ax.contourf(x[i], y[j], pdf[numpy.ix_(j, i)], levels, cmap=cmap,
+                       zorder=zorder, vmin=0, vmax=pdf.max(), *args, **kwargs)
     for c in cbar.collections:
         c.set_cmap(cmap)
 
-    ax.contour(x[i], y[j], pdf[numpy.ix_(j, i)], contours, zorder=zorder,
-               vmin=0, vmax=1.2, linewidths=0.5, colors='k', *args, **kwargs)
-    ax.patches += [plt.Rectangle((0, 0), 1, 1, fc=cmap(0.999), ec=cmap(0.33),
+    ax.contour(x[i], y[j], pdf[numpy.ix_(j, i)], levels, zorder=zorder,
+               vmin=0, vmax=pdf.max(), linewidths=linewidths, colors='k',
+               *args, **kwargs)
+    ax.patches += [plt.Rectangle((0, 0), 1, 1, fc=cmap(0.999), ec=cmap(0.32),
                                  lw=2, label=label)]
 
     ax.set_xlim(*check_bounds(x[i], xmin, xmax), auto=True)
     ax.set_ylim(*check_bounds(y[j], ymin, ymax), auto=True)
     return cbar
 
 
+def hist_plot_2d(ax, data_x, data_y, *args, **kwargs):
+    """Plot a 2d marginalised distribution as a histogram.
+
+    This functions as a wrapper around matplotlib.axes.Axes.hist2d
+
+    Parameters
+    ----------
+    ax: matplotlib.axes.Axes
+        axis object to plot on
+
+    data_x, data_y: numpy.array
+        x and y coordinates of uniformly weighted samples to generate kernel
+        density estimator.
+
+    xmin, xmax, ymin, ymax: float
+        lower/upper prior bounds in x/y coordinates
+        optional, default None
+
+    levels: list
+        amount of mass within each iso-probability contour.
+        If None defaults to hist2d.
+        optional, default None
```

Comment:
Maybe good to specify that it defaults to `matplotlib`'s `hist2d`?

Reply:
Made this a little more descriptive in the latest push.